src/authGenerator.js

```diff
@@ -0,0 +1,99 @@
+const JWT = require('jsonwebtoken');
+
+const Boom = require('@hapi/boom');
+const crypto = require('crypto');
+const util = require('util');
+
+const randomBytes = util.promisify(crypto.randomBytes);
+const pbkdf2 = util.promisify(crypto.pbkdf2);
+let sequelize;
+let authConfig;
+exports.authGenerator = async (request, options) => {
+    authConfig = options.config.authentication;
+    let { identityKey, passcodeKey, authModel } = authConfig;
+    identityKey = identityKey || 'username';
+    passcodeKey = passcodeKey || 'password';
+    authModel = authModel || 'User';
+    if (request.payload && request.payload[identityKey] && request.payload[passcodeKey]) {
+        sequelize = options.sequelize;
+        const queryOptions = {
+            where: {},
+        };
+        const username = request.payload[identityKey];
+        const password = request.payload[passcodeKey];
+
+        queryOptions.where[identityKey] = username;
+
+        try {
+            const record = await sequelize.models[authModel].findOne(queryOptions);
+            if (record) {
+                if (exports.verifyPassword(password, record[passcodeKey])) {
+                    const jwt = JWT.sign(
+                        { id: record.id, username: record.firstName, scope: 'admin' },
```

Comment:
We should not hardcode the scope to admin. Lets have it remove till we implement authorisation